File: LinkedList/DLL/insert a node (atHead)/insertNode.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

func insertAtHead(head *ListNode, X int) *ListNode {
	return &ListNode{Val: X, Next: head}

	//this is how you intilaise a struct in go, it is equivalent to

	/*
	    var temp *ListNode          // declaration only
	temp = &ListNode{Val: X}    // assigned later

	*/
}

File: LinkedList/DLL/insert a node (atHead)/insertNode.ts
/*Definiton of singly Linked List
class ListNode {
    constructor(val = 0, next = null) {
        this.val = val;
        this.next = next;
    }
}
*/

class Solution {
    insertAtHead(head, X) {

        let temp = ListNode(X);
        temp.next = head;
        return temp;

    }
}